user-service/controller/user-controller.js

```diff
@@ -34,15 +38,79 @@ export async function signIn(req, res) {
         const { username, password } = req.body;
         const user = await _FindOneuser(username,password)
         if (user) {
-            const accessToken = jwt.sign({username:username},ACCESS_TOKEN_SECRET)
-            console.log(accessToken)
-            // res.json({accessToken:accessToken})
-            res.cookie("accesstoken",accessToken,{httpOnly:true})
-            return res.status(200).json({message: `Log in is successful!`,accesstoken:accessToken });
+            const accessToken = jwt.sign({username:username},ACCESS_TOKEN_SECRET,{expiresIn:"1h"})
+            const refreshToken = jwt.sign({username:username},REFRESH_ACCESS_TOKEN_SECRET,{expiresIn:"1h"})
+            user.refreshToken = refreshToken
+            user.save()
+            //refresh token in cookie
+            //access token in data
+            if(validatePassword(password, user.password)){
+                return res.cookie("refreshtoken",refreshToken,{httpOnly:true}).status(200).json({message: `Log in is successful!`,accesstoken:accessToken });
+            }  else {
+                return res.status(400).json({message: 'Incorrect username/password!'});
+            }
+            
         } else {
             return res.status(400).json({message: 'Incorrect username/password!'});
         }
     } catch (err) {
         return res.status(500).json({message: 'Database failure when trying to log in!'})
     }
 }
+
+
+
+export async function logout(req, res) {
+    try {
+        // Get token from request
+        const accessToken = req.accessToken
+        if (!accessToken) {
+            return res.status(401).json({message: 'Token is missing or invalid!'});
+        }
+        const resp = await _addBlackList(accessToken);
+        if (resp.err) {
+            return res.status(409).json({message: 'Unable to add to blacklist'});
+        } else {
+            return res.status(200).json({ message: "User logout is successful" })
+        }
+    } catch (err) {
+        return res.status(500).json({message: 'Database failure when logging out!'})
+    }
+}
+
+export async function updateUser(req, res) {
+    try {
+        const { username } = req.user
+        const token = req.accessToken
+        const resp = await _FindUserbyUsername(username)
+        if (resp.err) {
+            logger.error(resp.err)
+            return res.status(400).json({ message: "Could not get user info" })
+        }
+        const user = resp[0]
+        let passwordUpdated = false
+        if (req.body?.oldPassword && req.body?.newPassword) {
+            if (req.body.oldPassword === req.body.newPassword) {
+                return res.status(400).json({ message: "Previous password and updated password cannot be the same" })
+            }
+            if(!validatePassword(req.body.oldPassword, user.password)){
+                console.log("incorrect")

```

Comment:
Is this console log required? If not please delete, if yes can you help make it clearer?